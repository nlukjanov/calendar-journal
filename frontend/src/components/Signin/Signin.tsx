import React, { useState, useEffect } from 'react';
import { useHistory } from 'react-router-dom';
import axios from 'axios';
import Form from 'react-bootstrap/Form';
import Button from 'react-bootstrap/Button';
import Card from 'react-bootstrap/Card';
import { setToken } from '../../lib/authHelper';

const Signin = () => {
  const history = useHistory();
  const [formData, setFormData] = useState();
  const [signinErrors, setSigninErrors] = useState();

  const handleChange = ({ target: { name, value } }) => {
    const formNewData = { ...formData, [name]: value };
    setFormData(formNewData);
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      const res = await axios.post('http://localhost:4000/api/login', formData);
      setToken(res.data.token);
      history.push('/myjournal');
      // add notification with res
    } catch (error) {
      setSigninErrors({ errors: error.response.data.message });
    }
  };

  useEffect(() => {
    console.log('error log', signinErrors?.errors);
  }, [signinErrors]);

  return (
    <Card className='mx-auto mt-5 border-light' style={{ maxWidth: '25rem' }}>
      <Form onSubmit={handleSubmit}>
        <Form.Group controlId='email'>
          <Form.Label>Email</Form.Label>
          <Form.Control
            name='email'
            type='email'
            placeholder='Enter email'
            onChange={handleChange}
          />
          <Form.Text className='text-muted'>
            We never share your email with anyone else.
          </Form.Text>
        </Form.Group>

        <Form.Group controlId='password'>
          <Form.Label>Password</Form.Label>
          <Form.Control
            type='password'
            name='password'
            placeholder='Password'
            onChange={handleChange}
            autoComplete='on'
          />
        </Form.Group>
        {signinErrors && (
          <div>
            <small>{signinErrors.errors}</small>
          </div>
        )}
        <Button variant='primary' type='submit'>
          Sign In
        </Button>
      </Form>
    </Card>
  );
};

export default Signin;
