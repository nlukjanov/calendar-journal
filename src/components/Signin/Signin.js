import React, { useState, useEffect } from 'react';
import { useHistory } from 'react-router-dom';
import axios from 'axios';
import { setToken } from '../../lib/authHelper';

const Signin = () => {
  const history = useHistory();
  const [formData, setFormData] = useState();
  const [signinErrors, setSigninErrors] = useState();

  const handleChange = ({ target: { name, value } }) => {
    const formNewData = { ...formData, [name]: value };
    setFormData(formNewData);
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      const res = await axios.post('http://localhost:4000/api/login', formData);
      setToken(res.data.token);
      history.push('/myjournal');
      // add notification with res
    } catch (error) {
      setSigninErrors({ errors: error.response.data.message });
    }
  };

  useEffect(() => {
    console.log('error log', signinErrors?.errors);
  }, [signinErrors]);

  return (
    <div>
      <form onSubmit={handleSubmit}>
        <div>
          <label htmlFor='email'>Email</label>
          <input
            id='email'
            type='text'
            name='email'
            placeholder='Email'
            onChange={handleChange}
          />
        </div>
        <div>
          <label htmlFor='password'>Password</label>
          <input
            id='password'
            type='password'
            name='password'
            placeholder='Password'
            autoComplete='on'
            onChange={handleChange}
          />
          {signinErrors && <small>{signinErrors.errors}</small>}
        </div>
        <div>
          <button type='submit'>Sign In</button>
        </div>
      </form>
    </div>
  );
};

export default Signin;
