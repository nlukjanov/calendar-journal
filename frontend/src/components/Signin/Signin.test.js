import React from 'react';

import { render } from '@testing-library/react';
import Signin from './Signin';

test('renders new journal entry', () => {
  const { container } = render(<Signin />);
  expect(container).toMatchSnapshot();
});
