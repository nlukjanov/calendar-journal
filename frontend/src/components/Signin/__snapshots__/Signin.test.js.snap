// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders new journal entry 1`] = `
<div>
  <div
    class="mx-auto mt-5 border-light card"
    style="max-width: 25rem;"
  >
    <form
      class=""
    >
      <div
        class="form-group"
      >
        <label
          class="form-label"
          for="email"
        >
          Email
        </label>
        <input
          class="form-control"
          id="email"
          name="email"
          placeholder="Enter email"
          type="email"
        />
        <small
          class="text-muted form-text"
        >
          We never share your email with anyone else.
        </small>
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
          for="password"
        >
          Password
        </label>
        <input
          autocomplete="on"
          class="form-control"
          id="password"
          name="password"
          placeholder="Password"
          type="password"
        />
      </div>
      <button
        class="btn btn-primary"
        type="submit"
      >
        Sign In
      </button>
    </form>
  </div>
</div>
`;
